### DISCUSSION TITLE
Where is the search app running?

### DISCUSSION THREAD

**Discussion by santafoo-aarini** [saleor/apps#943] [Q&A] [answered]
Hi there, 

A newbie question here. 
If I install the Search Saleor app via the dashboard, where does it run given the scenarios below?

**1. On products import from the app dashboard**
This one is clear. It runs client-side as stated there.

**2. On webhooks triggered (i.e. product create/update/delete)**
I presume not client-side. Imagine that a product is updated in the background, e.g. by another third-party app. 
So where does the app run then? In Saleor core? Or is the search app deployed on a server somewhere?

Please clarify. Thanx.

Cheers,

Jan

**Comment by lkostrowski** [comment, marked as answer]
Hey

If you install app via Saleor Dashboard in Saleor Cloud, you install the app hosted on Saleor infra. The app works out of the box and the Saleor team maintains it

The app is also open source, so on-premise users can use the app as well. In such scenario, you need to host the app on your own (e.g. on Vercel).

**Reply by santafoo-aarini** [reply]
Aaaah, I see, clear! Thanks for the explanation.

If I go for the on-premise solution in AWS, do you have any suggestion which aws resource works the best for the search app? EC2, Lambda fns??

**Reply by lkostrowski** [reply]
All apps are regular Next.js services, so natural choice is to host them on Vercel or similar platform.

You can look for resources on hosting next.js apps.

I guess it depends on your traffic. Next.js app can be deployed as a Dockerized service running on EC2 or K8S

you can see this https://github.com/saleor/apps/pull/537
